lib/dev.ts

```diff
@@ -17,16 +17,30 @@ var appDevOptions: null | DevOptions = null;
 let io: null | IoServer = null;
 let fileWatcher: null | Gaze = null;
 
+export const DEFAULT_GATEWAY = {
+	enabled: true,
+	bundleUrl: "",
+	tagName: DEFAULT_TAG_NAME,
+};
+
 export type DevOptions = {
   port?: number; // port to run dev server
   hot?: boolean; // enable hot reloading
   open?: boolean; // open browser
   network?: Network; // network to use
+
+	// TODO: maybe let it be only string?
```

Comment:
I'm thinking about this:

Since bundleUrl + tagName are an essential combo, and -g ${url} in commands has been a bit complicated, we can treat the gateway configuration in bos.config.json as the only place to configure and as source of truth. 

We do need to handle --no-gateway, which Commander.js (cli-tool) automatically assigns to the "gateway" variable in dev options. 

In short, I'm thinking gateway should be only boolean that handles whether it is enabled or not.

Then bos.config.json gateway handles the bundleUrl and tagName (and does not have enabled field, otherwise will have conflicts and unexpected behavior)

Reply:
So we will change README to reflect and say customizing gateway happens in bos.config.json

I'm not worried about backwards compatibility here, because only our projects are using custom gateway atm.

Reply:
So the requirements are:

1) If -g is false or there's the option --no-gateway, gateway is disabled;
2) if -g is true and there's a gateway configuration in `bos.config.json` takes the values from there;
3) if -g is true but there's no gateway configuration in `bos.config.json` use default values;

Question: shall we still make both `bundleUrl` and `tagName` both mandatory in `bos.config.json`?